### PR TITLE
Supporting scala 2.13 for spark 3.2.0

### LINKED ISSUE #1815: Provide Scala 2.13 builds
Now that the update to Spark 3.2 (which is cross-built for Scala 2.13) is in, it would be great to see 2.13 builds of elasticsearch-spark. I have been using a custom 7.12.0 build with Spark 3.2 and Scala 2.13.5 for a while now, which didn't require too many changes (https://github.com/Biomatters/elasticsearch-hadoop/commit/3b765befeb8abdacbfa06728a795f80e44445b9a), but it was quick and dirty and minimal in scope, so I imagine there might be more to doing it properly and with due diligence.

### PR DESCRIPTION
**Pull request by masseyke**
This commit adds support for spark 3.2 running scala 2.13.
Closes #1815

### CODE REVIEW

File: spark/core/src/main/scala/org/elasticsearch/spark/rdd/JDKCollectionConvertersCompat.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+package org.elasticsearch.spark.rdd
+
+/**
+ * Magic to get cross-compiling access to scala.jdk.CollectionConverter
+ * with a fallback on scala.collection.JavaConverters, without deprecation
+ * warning in any Scala version.
+ * From https://github.com/scala/scala-collection-compat/issues/208#issuecomment-497735669
+ */
+private[rdd] object JDKCollectionConvertersCompat {
```
Comment (izhangzhihao) [inline spark/core/src/main/scala/org/elasticsearch/spark/rdd/JDKCollectionConvertersCompat.scala:27]: Why not using `import scala.jdk.CollectionConverters._` introduced by [Backport scala.jdk.CollectionConverters to 2.11 / 2.12](https://github.com/scala/scala-collection-compat/pull/217)

Comment (masseyke) [reply spark/core/src/main/scala/org/elasticsearch/spark/rdd/JDKCollectionConvertersCompat.scala:27]: Thanks, I didn't know about that. We unfortunately also support scala 2.10 though -- do you know if it has been backported to 2.10?

Comment (izhangzhihao) [reply spark/core/src/main/scala/org/elasticsearch/spark/rdd/JDKCollectionConvertersCompat.scala:27]: It seems not available for 2.10

**Review by izhangzhihao — APPROVED** [review]

**Review by jbaiera — APPROVED** [review]
JDKCollectionConvertersCompat 🤯

**Pull request merged**